streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -342,7 +342,19 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
 
         maybeThrowTaskExceptions(taskCloseExceptions);
 
-        createNewTasks(activeTasksToCreate, standbyTasksToCreate);
+        rebalanceInProgress = true;
```

Comment:
If we set `rebalanceInProgress = true`, it seems the later call to `commitTasksAndMaybeUpdateCommittableOffsets` would exit early and return `-1`? Does not seem right?

I cannot remember the full purpose of `rebalanceInProgress` flag, and given cooperator rebalancing and processing record during a rebalance, wondering if semantics actually changed?

Edit: just see that there is a discussion about this below.

Reply:
Yeah. this one was suggested by Guozhang. It is still not passing all the tests though.